lib/Mitigation.pm

```diff
@@ -0,0 +1,352 @@
+# SUSE's openQA tests
+#
+# Copyright © 2009-2013 Bernhard M. Wiedemann
+# Copyright © 2012-2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Mitigation testcase library provide a class to execute
+# common test steps. Specifiy checkpoint could be done in indiviaul
+# test module via override functions in this class, or extent the test.
+#
+# Usage, when you need to testing a mitigation function.
+# you have to inital a hash struct seems like:
+#
+#my %mitigations_list =
+#  (
+#    name                   => "l1tf",
+#    CPUID                  => hex '10000000',
+#    IA32_ARCH_CAPABILITIES => 8,                #bit3 --SKIP_L1TF_VMENTRY
+#    parameter              => 'l1tf',
+#    cpuflags               => ['flush_l1d'],
+#    sysfs_name             => "l1tf",
+#    sysfs                  => {
+#        "full"         => "Mitigation: PTE Inversion; VMX: cache flushes, SMT disabled",
+#        "full,force"   => "Mitigation: PTE Inversion; VMX: cache flushes, SMT disabled",
+#        "flush"        => "Mitigation: PTE Inversion; VMX: conditional cache flushes, SMT vulnerable",
+#        "flush,nosmt"  => "Mitigation: PTE Inversion; VMX: conditional cache flushes, SMT disabled",
+#        "flush,nowarn" => "Mitigation: PTE Inversion; VMX: conditional cache flushes, SMT vulnerable",
+#        "off"          => "Mitigation: PTE Inversion; VMX: vulnerable",
+#        "default"      => "Mitigation: PTE Inversion; VMX: conditional cache flushes, SMT vulnerable",
+#    },
+#    cmdline => [
+#        "full",
+#        "full,force",
+#        "flush",
+#        "flush,nosmt",
+#        "flush,nowarn",
+#        "off",
+#    ],
+#  );
+#
+#  "name", the name of test module or mitigations name.
+#  "CPUID", the bit that should be check via cpuid instruction. follow Intel manual.
+#  "IA32_ARCH_CAPABILITIES", the bit that means this mitigations might be fixed by Hardware.
+#  "parameter", the name of parameter on kernel cmdline, as switch to enable or disable this mitigation in system.
+#  "cpuflags", the flag[s] name should be appear at /proc/cpuinfo or lscpu.
+#  "sysfs_name", the name of entry in sysfs: /sys/devices/system/cpu/vulnerabilities/*,
+#  "sysfs", a hash with  {"name" => "string"}
+#  	Means, when "name" be used as a kernel parameter to value, the "string" should be the context of "sysfs_name".
+#  "cmdline", an array that store what kernel parameter will be tested.
+#
+#TODO:
+#In the future, there will be a dmesg item, it should seems like "sysfs", that could be used for check the some messages in dmesg log.
```

Comment:
Please don't add TODO in code. Better leave it out and keep in your local git repository, either in `git stash` or a temporary "WIP"-commit and use tickets for planning

Reply:
OK. I will remove them.